From fa82974d8bd1fded78e630ab71a7ded1b11d0e33 Mon Sep 17 00:00:00 2001
From: Adrian Freihofer <adrian.freihofer@scs.ch>
Date: Wed, 14 Feb 2018 22:01:36 +0100
Subject: [PATCH 13/15] Install iiod upstart conf file

If WITH_UPSTART is set to ON the upstart conf file of iiod gets
installed into ${UPSTART_CONF_INSTALL_DIR} folder.
UPSTART_CONF_INSTALL_DIR defaults to /etc/init.

Signed-off-by: Adrian Freihofer <adrian.freihofer@gmail.com>
---
 CMakeLists.txt                             | 3 +++
 iiod/CMakeLists.txt                        | 5 +++++
 iiod/init/{iiod.conf => iiod.conf.cmakein} | 2 +-
 3 files changed, 9 insertions(+), 1 deletion(-)
 rename iiod/init/{iiod.conf => iiod.conf.cmakein} (83%)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 32a22ad..a115e87 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -411,6 +411,9 @@ if(WITH_IIOD)
 	option(WITH_SYSVINIT "Enable installation of SysVinit script for iiod" OFF)
 	set(SYSVINIT_INSTALL_DIR /etc/init.d CACHE PATH "default install path for SysVinit scripts")
 
+	option(WITH_UPSTART "Enable installation of upstart config file for iiod" OFF)
+	set(UPSTART_CONF_INSTALL_DIR /etc/init CACHE PATH "default install path for upstart conf files")
+
 	if (NOT PTHREAD_LIBRARIES)
 		message(WARNING "IIOD requires threads support; disabling")
 		set(WITH_IIOD OFF CACHE BOOL "" FORCE)
diff --git a/iiod/CMakeLists.txt b/iiod/CMakeLists.txt
index 8947601..ab30605 100644
--- a/iiod/CMakeLists.txt
+++ b/iiod/CMakeLists.txt
@@ -74,3 +74,8 @@ if (WITH_SYSVINIT)
 	        PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ
 	        DESTINATION ${SYSVINIT_INSTALL_DIR})
 endif()
+
+if (WITH_UPSTART)
+	configure_file(${CMAKE_CURRENT_SOURCE_DIR}/init/iiod.conf.cmakein ${PROJECT_BINARY_DIR}/init/iiod.conf)
+	install(FILES ${PROJECT_BINARY_DIR}/init/iiod.conf DESTINATION ${UPSTART_CONF_INSTALL_DIR})
+endif()
diff --git a/iiod/init/iiod.conf b/iiod/init/iiod.conf.cmakein
similarity index 83%
rename from iiod/init/iiod.conf
rename to iiod/init/iiod.conf.cmakein
index cb91b1c..1ac3c88 100644
--- a/iiod/init/iiod.conf
+++ b/iiod/init/iiod.conf.cmakein
@@ -10,4 +10,4 @@ stop on runlevel [!2345]
 
 respawn
 
-exec /usr/sbin/iiod -D
+exec @CMAKE_INSTALL_FULL_SBINDIR@/iiod -D
-- 
2.11.0

